embassy-net-driver/src/lib.rs

```diff
@@ -4,6 +4,16 @@
 
 use core::task::Context;
 
+/// Representation of an hardware address, such as an Ethernet address or an IEEE802.15.4 address.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum HardwareAddress {
```

Comment:
could you add an `Ip` variant with no contents, like in smoltcp? otherwise drivers with IP medium have to return a "dummy" hardware address which is a bit awkward.

(I know this was already "wrong" before, but now that you're changing this it would be the ideal time to fix it. Thank you! :pray: )

Reply:
```diff
diff --git a/embassy-net/src/lib.rs b/embassy-net/src/lib.rs
index 746acae4..dbefb0ef 100644
--- a/embassy-net/src/lib.rs
+++ b/embassy-net/src/lib.rs
@@ -252,23 +252,7 @@ impl<D: Driver + 'static> Stack<D> {
         resources: &'static mut StackResources<SOCK>,
         random_seed: u64,
     ) -> Self {
-        #[cfg(any(feature = "medium-ethernet", feature = "medium-ieee802154"))]
-        let medium = device.capabilities().medium;
-
-        let hardware_addr = match medium {
-            #[cfg(feature = "medium-ethernet")]
-            Medium::Ethernet => to_smoltcp_hardware_address(device.hardware_address()),
-            #[cfg(feature = "medium-ip")]
-            Medium::Ip => HardwareAddress::Ip,
-            #[cfg(feature = "medium-ieee802154")]
-            Medium::Ieee802154 => to_smoltcp_hardware_address(device.hardware_address()),
-            #[allow(unreachable_patterns)]
-            _ => panic!(
-                "Unsupported medium {:?}. Make sure to enable it in embassy-net's Cargo features.",
-                medium
-            ),
-        };
-        let mut iface_cfg = smoltcp::iface::Config::new(hardware_addr);
+        let mut iface_cfg = smoltcp::iface::Config::new(to_smoltcp_hardware_address(device.hardware_address()));
         iface_cfg.random_seed = random_seed;

         let iface = Interface::new(
```

Including this, I suppose.

Reply:
yup! :D